Tests/Tests/PromiseKit/XCTestCaseTests+Promise.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  XCTestCaseTests+Promise.swift
+//  PovioKit
+//
+//  Created by Egzon Arifi on 27/09/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import XCTest
+import PovioKitPromise
+
+class XCTestCaseTests_Promise: XCTestCase {
+  func test_awaitPromise_returnsSuccessResultWithExpectedObject() throws {
+    let anyObject = anyObject()
+    
+    switch try awaitPromise(anySuccessPromise(anyObject)) {
+    case .success(let object):
+      XCTAssertEqual(anyObject, object)
+    case .failure:
+      XCTFail("Expected success result")
+    }
+  }
+  
+  func test_awaitPromise_returnsFailureResultWithExpectedError() throws {
+    let anyError = anyNSError()
+    
+    let failurePromise: Promise<TestDouble> = anyFailurePromise(anyError)
+    switch try awaitPromise(failurePromise) {
+    case .failure(let error as NSError):
+      XCTAssertEqual(anyError as NSError, error)
+    case .success:
+      XCTFail("Expected failure result")
+    }
```

Comment:
We could refactor this with a helper method `expect`

```swift
  func expect<T: Equatable>(_ promise: Promise<T>, toCompleteWith expectedResult: Result<T, Error>, file: StaticString = #file, line: UInt = #line) throws {
    switch try (awaitPromise(promise), expectedResult) {
    case let (.success(success), .success(expectedSuccess)):
      XCTAssertEqual(success, expectedSuccess, "Expected \(expectedSuccess) got \(success) instead", file: file, line: line)
    case let (.failure(error as NSError), .failure(expectedError as NSError)):
      XCTAssertEqual(error, expectedError, "Expected \(expectedError) got \(error) instead", file: file, line: line)
    default:
      XCTFail("Expected \(expectedResult)", file: file, line: line)
    }
  }
```
Then this test could look like: 
```swift
    let anyError = anyNSError()
    let failurePromise: Promise<TestDouble> = anyFailurePromise(anyError)
    
    try expect(failurePromise, toCompleteWith: .failure(anyError))
```

The `test_awaitPromise_returnsSuccessResultWithExpectedObject` test would look like: 
```swift
    let anyObject = anyObject()
    let successPromise = anySuccessPromise(anyObject)
    
    try expect(successPromise, toCompleteWith: .success(anyObject))
```